testsupport/nexus-ldap-testsupport/src/main/java/org/sonatype/ldaptestsuite/LdapServer.java

```diff
@@ -570,89 +221,141 @@ public void dispose() {
         directoryService.shutdown();
       }
       catch (Exception e) {
-        System.out.println("Failed to stop directoryService");
+        log.error("Failed to stop directoryService", e);
       }
     }
+    return this;
   }
 
-  public void stop()
-      throws StoppingException
-  {
-    try {
-      dispose();
-    }
-    catch (Exception e) {
-      throw new StoppingException("Error stopping embedded ApacheDS server.", e);
-    }
+  public boolean isStarted() {
+    return directoryService != null && directoryService.isStarted();
   }
 
-  public void enableLogging(Logger logger) {
-    this.logger = logger;
+  /**
+   * Returns the actual port LDAP server is using.
+   */
+  public int getPort() {
+    return port;
   }
 
   /**
-   * Imports the LDIF entries packaged with the Eve JNDI provider jar into the newly created system partition to
-   * prime
-   * it up for operation. Note that only ou=system entries will be added - entries for other partitions cannot be
-   * imported and will blow chunks.
-   *
-   * @param in the input stream with the ldif
-   * @throws NamingException if there are problems reading the ldif file and adding those entries to the system
-   *                         partition
+   * Returns the configuration.
    */
-  protected void importLdif(InputStream in)
-      throws NamingException
-  {
+  public LdapServerConfiguration getConfiguration() {
+    return configuration;
+  }
+
+  // ==
+
+  @VisibleForTesting
+  protected DirectoryService getDirectoryService() {
+    return directoryService;
+  }
+
+  // ==
+
+  private void deleteWorkingDirectory() {
     try {
-      for (LdifEntry ldifEntry : new LdifReader(in)) {
-        rootDSE.add(new DefaultServerEntry(rootDSE.getDirectoryService().getRegistries(),
-            ldifEntry.getEntry()));
-      }
+      Files.walkFileTree(configuration.getWorkingDirectory().toPath(), new SimpleFileVisitor<Path>()
```

Comment:
new Files api is indeed fun, but why use this verbose mechanism when we have         FileUtils.deleteDirectory(dir) and several other library mechanisms we can depend on for directory deletion?